### PR TITLE
git-hub-shorten-url: Support shorthand and code

### PR DESCRIPTION
**Pull request by leoj3n**
My proposed changes enhance the `git-hub-shorten-url` command so it can be used like:

```
~ ❯❯❯ git-hub-shorten-url sorin-ionescu/prezto instantly-awesome-zsh
http://git.io/instantly-awesome-zsh
```

:+1:

### CODE REVIEW

**Comment by devinrhode2** [comment]
hey @sorin-ionescu it seems that the 'new hub' is actually the same github/hub repository, still hub on homebrew. At the bottom of the readme on github/hub you can see "hub in ruby" (previous implementation) - also the latest commits (even by the creator of gh) are on the github/hub repo. Hey, my comment is almost one year since the last comment.

Regarding the word "slug" for urls - I learned it before I knew how to code thanks to wordpress.

File: modules/git/functions/git-hub-shorten-url [hunk 1]
```diff
@@ -2,21 +2,29 @@
...
+  return 1
+elif [[ "$url" == '-' ]]; then
+  read url <&0
+elif [[ "$url" != *github.com* ]]; then
+  url="https://github.com/$url"
 fi
 
 if (( $+commands[curl] )); then
-  curl -s -i 'http://git.io' -F "url=$url" | grep 'Location:' | sed 's/Location: //'
+  local result="$(curl -s -i 'http://git.io' -F "url=$url" -F "code=$2")"
```
Comment (belak) [inline modules/git/functions/git-hub-shorten-url]: This should also probably use https

**Comment by belak** [comment]
I merged a smaller PR which also modifies this so there are conflicts now. Sorry about that.

**Comment by leoj3n** [comment]
> Sorry, we're no longer accepting new links to shorten.

**Comment by belak** [comment]
Thanks for the update! I'll guess if there are no new links, we might as well drop this function.

**Comment by leoj3n** [comment]
> Thanks for the update! I'll guess if there are no new links, we might as well drop this function.

Shortening a URL (in general) is a nice feature, but probably not something needed on a daily basis.

I would axe it unless you think it's a good example that can be recreated using a still functioning service.

**Pull request closed without merging**